assets/js/index.js

```diff
@@ -160,18 +161,53 @@ document.addEventListener('DOMContentLoaded', () => {
     }
   }
 
-  startRecordButton.addEventListener('click', () => {
-    recordedChunks = []
-    mediaRecorder.start()
-    startRecordButton.disabled = true
-    stopRecordButton.disabled = false
-  })
-
-  stopRecordButton.addEventListener('click', () => {
-    mediaRecorder.stop()
-    startRecordButton.disabled = false
-    stopRecordButton.disabled = true
-  })
+  if (startRecordButton && stopRecordButton) {
+    startRecordButton.addEventListener('click', () => {
+      recordedChunks = []
+      mediaRecorder.start()
+      startRecordButton.disabled = true
+      stopRecordButton.disabled = false
+    })
+
+    stopRecordButton.addEventListener('click', () => {
+      mediaRecorder.stop()
+      startRecordButton.disabled = false
+      stopRecordButton.disabled = true
+    })
+
+    startVideoStream()
+  }
+})
 
-  startVideoStream()
+// get location and redirect to checkin page
+document.addEventListener('DOMContentLoaded', () => {
+  const submitCheckinButton = document.getElementById('submit-checkin-video')
+  const locationAlert = document.getElementById('enable-location-alert')
+  locationAlert.style.display = 'none'
+  if (submitCheckinButton) {
+    submitCheckinButton.addEventListener('click', event => {
+      event.preventDefault()
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(
+          position => {
+            const { latitude, longitude } = position.coords
+            const userLocationInput = document.getElementById('user-location')
```

Comment:
What is the user input location for? They've stated it verbally? Or are we using a different pattern for input here? input box with free text?

Reply:
We agree that in a pilot solution this would be posted back to an api rather than kept in a hidden field. Not to be fixed in this prototype.